src/main/java/org/mal_lang/langspec/ttc/TtcDistribution.java

```diff
@@ -16,568 +16,488 @@
+    }
+  },
+
+  /**
+   * Enum constant representing an exponential distribution.
+   *
+   * @since 1.0.0
+   */
+  EXPONENTIAL("Exponential") {
+    @Override
+    public void validateArguments(double... arguments) {
+      requireFinite(requireArgumentsSize(requireNonNull(arguments), 1));
+      double rate = arguments[0];
+      requirePositive(rate);
+    }
+
+    @Override
+    public double getMeanTtc(double... arguments) {
+      this.validateArguments(arguments);
+      double rate = arguments[0];
+      return 1.0 / rate;
+    }
+  },
+
+  /**
+   * Enum constant representing a gamma distribution.
+   *
+   * @since 1.0.0
+   */
+  GAMMA("Gamma") {
+    @Override
+    public void validateArguments(double... arguments) {
+      requireFinite(requireArgumentsSize(requireNonNull(arguments), 2));
+      double shape = arguments[0];
+      double scale = arguments[1];
+      requirePositive(shape);
+      requirePositive(scale);
+    }
+
+    @Override
+    public double getMeanTtc(double... arguments) {
+      this.validateArguments(arguments);
+      double shape = arguments[0];
+      double scale = arguments[1];
+      return shape * scale;
+    }
+  },
+
+  /**
+   * Enum constant representing a log-normal distribution.
+   *
+   * @since 1.0.0
+   */
+  LOG_NORMAL("LogNormal") {
+    @Override
+    public void validateArguments(double... arguments) {
+      requireFinite(requireArgumentsSize(requireNonNull(arguments), 2));
+      // double normalMean = arguments[0];
+      double normalStandardDeviation = arguments[1];
+      requirePositive(normalStandardDeviation);
+    }
+
+    @Override
+    public double getMeanTtc(double... arguments) {
+      this.validateArguments(arguments);
+      double normalMean = arguments[0];
+      double normalStandardDeviation = arguments[1];
+      return Math.exp(normalMean + normalStandardDeviation * normalStandardDeviation / 2.0);
+    }
+  },
+
+  /**
+   * Enum constant representing a pareto distribution.
+   *
+   * @since 1.0.0
+   */
+  PARETO("Pareto") {
+    @Override
+    public void validateArguments(double... arguments) {
+      requireFinite(requireArgumentsSize(requireNonNull(arguments), 2));
+      double minimumValue = arguments[0];
+      double shape = arguments[1];
+      requirePositive(minimumValue);
+      requirePositive(shape);
+    }
+
+    @Override
+    public double getMeanTtc(double... arguments) {
+      this.validateArguments(arguments);
+      double minimumValue = arguments[0];
+      double shape = arguments[1];
+      return shape > 1.0 ? (shape * minimumValue) / (shape - 1.0) : Double.MAX_VALUE;
+    }
+  },
+
+  /**
+   * Enum constant representing a truncated normal distribution.
+   *
+   * @since 1.0.0
+   */
+  TRUNCATED_NORMAL("TruncatedNormal") {
+    @Override
+    public void validateArguments(double... arguments) {
+      requireFinite(requireArgumentsSize(requireNonNull(arguments), 2));
+      // double mean = arguments[0];
```

Comment:
shouldn't we validate this?

Reply:
The `mean` parameter of a normal distribution is valid for all real numbers, so there is nothing to validate. It is validated to be finite (not ±`infinity` or `nan`) on the line above. I kept the comment as a reference so that we remember that `arguments[0]` is the `mean` parameter. This could perhaps be done more elegantly in a documentation comment instead.

Reply:
Yea clear up the comment a little bit and I'm happy.